server/monitor/src/main/java/org/apache/accumulo/monitor/EmbeddedWebServer.java

```diff
@@ -51,7 +51,7 @@ public EmbeddedWebServer(Monitor monitor, int port) {
     secure = requireForSecure.stream().map(conf::get).allMatch(s -> s != null && !s.isEmpty());
 
     connector = new ServerConnector(server, getConnectionFactories(conf, secure));
-    connector.setHost(monitor.getHostname());
+    connector.setHost(monitor.getAdvertiseAddress().getHost());
```

Comment:
EmbeddedWebServer is called at the beginning of the run method for the Monitor so
`monitor.getAdvertiseAddress()` will always return null and the EmbeddedWebServer will fail to start.

The same hostname lookup logic from the monitor could be added here: 
```suggestion
    String hostName = "";
    if (monitor.getAdvertiseAddress() == null) {
      try {
         hostName = InetAddress.getLocalHost().getHostName();
      } catch (UnknownHostException e) {
        LOG.error("Unable to get hostname", e);
      }
    } else {
      hostName = monitor.getAdvertiseAddress().getHost();
    }
    connector.setHost(hostName);
```